include/zephyr/sys/errno_private.h

```diff
@@ -8,6 +8,7 @@
 #define ZEPHYR_INCLUDE_SYS_ERRNO_PRIVATE_H_
 
 #include <zephyr/toolchain.h>
+#include <zephyr/types.h> /* For Z_THREAD_LOCAL */
```

Comment:
This include was needed since the macro definition was moved from `<zephyr/toolchain.h>` in #79094